ldpd/lde.c

```diff
@@ -1708,6 +1727,34 @@ on_get_label_chunk_response(uint32_t start, uint32_t end)
 		current_label_chunk = listtail(label_chunk_list);
 }
 
+void
+lde_free_label(uint32_t label)
+{
+	struct listnode     *node;
+	struct label_chunk  *label_chunk;
+	uint64_t		     pos;
+
+	node = listhead(label_chunk_list);
+
+	while (node) {
+		label_chunk = listgetdata(node);
+
+		if (label_chunk) {
+			if ((label < label_chunk->end)  && (label >= label_chunk->start)) {
+				pos = 1 << (label - label_chunk->start);
+				label_chunk->used_mask &= ~pos;
+				/*nobdy using this chunk..free it if its not crrent_label_chunk*/
```

Comment:
couple of spellings: "nobody", "it's", "current"

Reply:
sorry.. so many mistakes.. I will correct all.